### PR TITLE
Use the first name as preferred name when the display name is empty

### PR DESCRIPTION
**Pull request by fishilico**
In the database, the display name might be empty. Use the first name to avoid a Django validation error:

    django.core.exceptions.ValidationError: {'preferred_name': [u'Ce champ ne peut pas \xeatre vide.']}

### CODE REVIEW

**Comment by fishilico** [comment]
The import is still failing because of accounts with a display name but without a first name. Here is a new commit.

**Comment by fishilico** [comment]
Hum, some external accounts have neither a display name nor a first name. In order to keep them, what about making ``User.firstname`` "``blank=True``"?

File: xorgauth/management/commands/importaccounts.py [hunk 1]
```diff
@@ -59,6 +59,29 @@ def handle(self, *args, **options):
...
+            # from the email address
+            if not account_data['display_name']:
+                if account_data['firstname']:
+                    if is_verbose:
+                        print("... using first name (%r) as display name" % account_data['firstname'])
+                    account_data['display_name'] = account_data['firstname']
+                else:
+                    # This should only happen for external accounts
+                    if account_data['type'] != 'xnet':
+                        raise CommandError("No display_name nor firstname for a non-external account")
```
Comment (vberger) [inline xorgauth/management/commands/importaccounts.py]: Nitpick : could be nice to dump `account_data` here in order to ease finding any faulty data in the future?

File: xorgauth/management/commands/importaccounts.py [hunk 2]
```diff
@@ -59,6 +59,29 @@ def handle(self, *args, **options):
...
+
+                    display_name = account_data['email'].split('@')[0]
+                    if not display_name:
+                        raise CommandError("No display_name nor firstname nor email in account data")
```
Comment (vberger) [inline xorgauth/management/commands/importaccounts.py]: Same here, dump `account_data`?

**Pull request merged**